### PR TITLE
Simplify COS Python API

### PR DESCRIPTION
**Pull request by anfeng**
This PR remove SQLContext from Python CaffeOnSpark constructor. It makes API consistent b/w Scala and Python.

We also change the return type of trainWithValidation() to DataFrame.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: caffe-grid/src/main/scala/com/yahoo/ml/caffe/CaffeOnSpark.scala [hunk 1]
```diff
@@ -323,38 +337,37 @@ class CaffeOnSpark(@transient val sc: SparkContext) extends Serializable {
...
             processor.solversFinished = !res
           }
-          Iterator(res)
+
+          if (index>0)
+            Iterator(null)
+          else
+            processor.validationBlobOutput.map{e => Row.fromSeq(e.toSeq)}.toIterator
         }
-      }.reduce(_ && _)
```
Comment (mriduljain) [inline caffe-grid/src/main/scala/com/yahoo/ml/caffe/CaffeOnSpark.scala]: Without an action...how does this work?

File: Makefile [hunk 2]
```diff
@@ -24,7 +25,7 @@ buildosx:
 	export DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH2}"; mvn -B package
 	cp -r ${CAFFE_ON_SPARK}/caffe-public/python/caffe ${CAFFE_ON_SPARK}/caffe-grid/src/main/python/
 	cd ${CAFFE_ON_SPARK}/caffe-grid/src/main/python/; zip -r caffeonsparkpythonapi  *; mv caffeonsparkpythonapi.zip ${CAFFE_ON_SPARK}/caffe-grid/target/; cd ${CAFFE_ON_SPARK}
-	${CAFFE_ON_SPARK}/caffe-grid/src/test/python/PythonTest.sh
+	export LD_LIBRARY_PATH="${LD_LIBRARY_PATH2}"; export SPARK_HOME="${SPARK_HOME}"; ${CAFFE_ON_SPARK}/caffe-grid/src/test/python/PythonTest.sh
```
Comment (mriduljain) [inline Makefile]: I think I pasted this incorrectly. It should be export DYLD_LIBARARY_PATH="${LD_LIBRARY_PATH2}

**Comment by mriduljain** [comment]
+1

**Pull request merged**